[https://nvbugs/5412562][feat] Allocate MoE workspace only when necessary

nv-yilinf opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit


## Summary by CodeRabbit

* **Refactor**
  * Improved internal workspace management for better efficiency and reduced unnecessary memory allocations. No changes to user-facing features or functionality.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

<!--
Please write the PR title by following this template:

[JIRA ticket/NVBugs ID/GitHub issue][fix/feat/doc/infra/...] \<summary of this PR\>

For example, assume I have a PR to support a new feature about cache manager for JIRA ticket TRTLLM-1000, it would be like:

[TRTLLM-1000][feat] Support a new feature about cache manager

Or I have a PR to fix a Llama3 accuracy issue:

[https://nvbugs/1234567][fix] Fix Llama3 accuracy issue
-->

## Description
In current MoE runner implementation, the runner will allocate a new workspace Tensor each time invoking the kernel. Even though backed by torch's caching allocator, frequent cudaMalloc/Frees are usually considered not a good practice and [sometimes](https://nvbugspro.nvidia.com/bug/5412562) causes cudaMalloc to take ~100ms. 
In this PR we fix this issue by maintaining the workspace tensor as a Class variable and only reallocates when target size is larger than current size or when capturing cuda graph. 

## Test Coverage

<!--
Please list clearly what are the relevant test(s) that can safeguard the changes in the PR. This helps us to ensure we have sufficient test coverage for the PR.
-->

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--reuse-test (optional)pipeline-id --disable-fail-fast --skip-test --stage-list "A10-PyTorch-1, xxx" --gpu-type "A30, H100_PCIe" --test-backend "pytorch, cpp" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx" --detailed-log --debug(experimental)]`

Launch build/test pipelines. All previously running jobs will be killed.

`--reuse-test (optional)pipeline-id ` *(OPTIONAL)* : Allow the new pipeline to reuse build artifacts and skip successful test stages from a specified pipeline or the last pipeline if no pipeline-id is indicated. If the Git commit ID has changed, this option will be always ignored. The DEFAULT behavior of the bot is to reuse build artifacts and successful test results from the last pipeline.

`--disable-reuse-test ` *(OPTIONAL)* : Explicitly prevent the pipeline from reusing build artifacts and skipping successful test stages from a previous pipeline. Ensure that all builds and tests are run regardless of previous successes.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-PyTorch-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-PyTorch-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--test-backend "pytorch, cpp"` *(OPTIONAL)* : Skip test stages which don't match the specified backends. Only support [pytorch, cpp, tensorrt, triton]. Examples: "pytorch, cpp" (does not run test stages with tensorrt or triton backend). Note: Does **NOT** update GitHub pipeline status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests in addition to running L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx".

`--detailed-log ` *(OPTIONAL)* : Enable flushing out all logs to the Jenkins console. This will significantly increase the log volume and may slow down the job.

`--debug ` *(OPTIONAL)* : **Experimental feature**. Enable access to the CI container for debugging purpose. Note: Specify exactly one stage in the `stage-list` parameter to access the appropriate container environment. Note: Does **NOT** update GitHub check status.

For guidance on mapping tests to stage names, see `docs/source/reference/ci-overview.md`
and the `scripts/test_to_stage_mapping.py` helper.

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

tensorrt-cicd commented:
[PR_Github #13713](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/13713/) [ run ] completed with state `FAILURE`
[/LLM/main/L0_MergeRequest_PR pipeline #10303](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/10303/) completed with status: 'FAILURE'

nv-yilinf commented:
/bot run

tensorrt-cicd commented:
[PR_Github #13732](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/13732/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #13732](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/13732/) [ run ] completed with state `FAILURE`
[/LLM/main/L0_MergeRequest_PR pipeline #10319](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/10319/) completed with status: 'FAILURE'

nv-yilinf commented:
/bot run

nv-yilinf commented:
/bot run

nv-yilinf commented:
/bot run

tensorrt-cicd commented:
[PR_Github #14018](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14018/) [ run ] completed with state `FAILURE`
[/LLM/main/L0_MergeRequest_PR pipeline #10566](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/10566/) completed with status: 'FAILURE'

nv-yilinf commented:
/bot run

tensorrt-cicd commented:
[PR_Github #14172](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14172/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #14172](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14172/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #10698](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/10698/) completed with status: 'FAILURE'

nv-yilinf commented:
/bot run

nv-yilinf commented:
/bot run

tensorrt-cicd commented:
[PR_Github #14373](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14373/) [ run ] triggered by Bot

yiqingy0 committed:
[None][chore] Bump version to 1.1.0rc0 (#6651)

Signed-off-by: Yiqing Yan <yiqingy@nvidia.com>

amitz-nv committed:
[TRTLLM-6683][feat] Support LoRA reload CPU cache evicted adapter (#6510)

Signed-off-by: Amit Zuker <203509407+amitz-nv@users.noreply.github.com>

ruodil committed:
[None][test] correct test-db context for perf yaml file (#6686)

Signed-off-by: ruodil <200874449+ruodil@users.noreply.github.com>

hlu1 committed:
[None] [feat] Add model gpt-oss (#6645)

Signed-off-by: Hao Lu <14827759+hlu1@users.noreply.github.com>

xinhe-nv committed:
[https://nvbugs/5409414][fix] fix Not registered specs (#6660)

Signed-off-by: Xin He (SW-GPU) <200704525+xinhe-nv@users.noreply.github.com>
Co-authored-by: Larry <197874197+LarryXFly@users.noreply.github.com>

peaceh-nv committed:
[None][feat] : Add FP8 context MLA support for SM120 (#6059)

Signed-off-by: peaceh <103117813+peaceh-nv@users.noreply.github.com>

shaharmor98 committed:
[TRTLLM-6092][doc] Add LoRA feature usage doc (#6603)

Signed-off-by: Shahar Mor <17088876+shaharmor98@users.noreply.github.com>

syuoni committed:
[TRTLLM-6409][feat] Enable guided decoding with speculative decoding (part 1: two-model engine) (#6300)

Signed-off-by: Enwei Zhu <21126786+syuoni@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #14373](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14373/) [ run ] completed with state `FAILURE`
[/LLM/main/L0_MergeRequest_PR pipeline #10863](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/10863/) completed with status: 'FAILURE'

pcastonguay committed:
[TRTLLM-6881][feat] Include attention dp rank info with KV cache events (#6563)

Signed-off-by: Patrice Castonguay <55748270+pcastonguay@users.noreply.github.com>

EmmaQiaoCh committed:
[None][infra] Fix guardwords (#6711)

Signed-off-by: qqiao <qqiao@nvidia.com>

yiqingy0 committed:
[None][package] Pin cuda-python version to >=12,<13 (#6702)

Signed-off-by: Yiqing Yan <yiqingy@nvidia.com>
Signed-off-by: Yanchao Lu <yanchaol@nvidia.com>
Co-authored-by: Yanchao Lu <yanchaol@nvidia.com>

nv-guomingz committed:
[None][doc] Add deployment guide section for VDR task (#6669)

Signed-off-by: nv-guomingz <137257613+nv-guomingz@users.noreply.github.com>

raayandhar committed:
[None][fix] disagg ctx pp4 + gen pp4 integ test (#6489)

Signed-off-by: raayandhar <rdhar@nvidia.com>
Signed-off-by: Raayan Dhar <58057652+raayandhar@users.noreply.github.com>

mikeiovine committed:
[None][feat] Clean up ngram auto mode, add max_concurrency to configs (#6676)

Signed-off-by: Mike Iovine <6158008+mikeiovine@users.noreply.github.com>

pcastonguay committed:
[None][chore] Remove py_executor from disagg gh team (#6716)

Signed-off-by: Patrice Castonguay <55748270+pcastonguay@users.noreply.github.com>

chenopis committed:
[https://nvbugs/5423962][fix] Address broken links (#6531)

tongyuantongyu committed:
[None][fix] Migrate to new cuda binding package name (#6700)

Signed-off-by: Yuan Tong <13075180+tongyuantongyu@users.noreply.github.com>

symphonylyh committed:
[https://nvbugs/5410687][fix] Hopper w4a8 groupwise MoE interleave (#6708)

Signed-off-by: Haohang Huang <31998628+symphonylyh@users.noreply.github.com>

Tabrizian committed:
[None][feat] Add NCCL Symmetric Integration for All Reduce (#4500)

Signed-off-by: Iman Tabrizian <10105175+tabrizian@users.noreply.github.com>

dcampora committed:
[TRTLLM-6785][feat] BREAKING CHANGE Enable TRTLLM sampler by default (#6216)

Signed-off-by: Daniel Campora <961215+dcampora@users.noreply.github.com>

xinhe-nv committed:
[TRTQA-2920][fix] Add failed cases into waives.txt (#6719)

Signed-off-by: xinhe-nv <200704525+xinhe-nv@users.noreply.github.com>

ruodil committed:
[TRTLLM-5252][test] add for mistral_small_3.1_24b perf test (#6685)

Signed-off-by: ruodil <200874449+ruodil@users.noreply.github.com>

StudyingShao committed:
[TRTLLM-6744][feat] Remove input_sf swizzle for module WideEPMoE (#6231)

Signed-off-by: Jiang Shao <91270701+StudyingShao@users.noreply.github.com>

zhanghaotong committed:
[None][fix] Fix unnecessary GPU synchronization in torch sampler caused by incorrect tensor reference (#6626)

Signed-off-by: 皓聪 <zhanghaotong.zht@alibaba-inc.com>
Co-authored-by: 皓聪 <zhanghaotong.zht@alibaba-inc.com>

syuoni committed:
[TRTLLM-6854][feat] Enable guided decoding with disagg serving (#6704)

Signed-off-by: Enwei Zhu <21126786+syuoni@users.noreply.github.com>

2ez4bz committed:
[TRTLLM-5252][fix] Propagate mapping to intermediate layers (#6611)

This commit propagates the mapping to intermediate layers to enable
tensor parallelism (amongst other things) in them.

It also fixes issues with a unit test for TP for pixtral, and adds it to a
test list.

Signed-off-by: William Zhang <133824995+2ez4bz@users.noreply.github.com>

ruodil committed:
[None][test] fix yml condition error under qa folder (#6734)

Signed-off-by: ruodil <200874449+ruodil@users.noreply.github.com>

chang-l committed:
[None][doc] Add doc for multimodal feature support matrix (#6619)

Signed-off-by: Chang Liu <9713593+chang-l@users.noreply.github.com>

limin2021 committed:
[TRTLLM-6898][feat] make fused_moe_cute_dsl work on blackwell (#6616)

Signed-off-by: Mindy Li <11663212+limin2021@users.noreply.github.com>

leslie-fang25 committed:
[https://nvbugs/5436461][infra] Adjust free_gpu_memory_fraction of test_eagle3 to prevent OOM on CI (#6631)

Signed-off-by: leslie-fang25 <leslief@nvidia.com>

yuxianq committed:
[None][refactor] Combine resmooth_to_fp8_e8m0 and transform_sf_into_required_layout (#6654)

Signed-off-by: Yuxian Qiu <142763828+yuxianq@users.noreply.github.com>

JunyiXu-nv committed:
[https://nvbugs/5437106][fix] Fix llama4 scout TRTLLM attn_backend (#6690)

Signed-off-by: Junyi Xu <junyix@nvidia.com>

lancelly committed:
[None][fix] Remove lock related typo in py_executor (#6653)

Signed-off-by: Lanyu Liao <lancelly@users.noreply.github.com>

zhengd-nv committed:
[None][feat] move kv cache measure into transfer session (#6633)

Signed-off-by: zhengd-nv <200704041+zhengd-nv@users.noreply.github.com>

chuangz0 committed:
[None][fix]revert kvcache transfer (#6709)

Signed-off-by: Chuang Zhu <111838961+chuangz0@users.noreply.github.com>

stnie committed:
[TRTLLM-6650][fix] Enhance CUDA graph + Beam search to correctly handle padding (#6665)

Signed-off-by: Stefan Niebler <82932102+stnie@users.noreply.github.com>

Wanli-Jiang committed:
[TRTLLM-6308][feat] Support Aggregate mode for phi4-mm (#6184)

Signed-off-by: Wanli Jiang <35160485+Wanli-Jiang@users.noreply.github.com>

mikeiovine committed:
[None][feat] Optimize CUDA graph memory usage for spec decode cases (#6718)

Signed-off-by: Mike Iovine <6158008+mikeiovine@users.noreply.github.com>

venkywonka committed:
[TRTLLM-7025] [infra] Reorganize CODEOWNERS to rectify `examples` mapping (#6762)

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

Fridah-nv committed:
[None][doc] Move AutoDeploy README.md to torch docs (#6528)

Signed-off-by: Frida Hou <201670829+Fridah-nv@users.noreply.github.com>
Signed-off-by: Suyog Gupta <41447211+suyoggupta@users.noreply.github.com>
Co-authored-by: Suyog Gupta <41447211+suyoggupta@users.noreply.github.com>

dongfengy committed:
[None][fix] WAR GPT OSS on H20 with Triton MOE (#6721)

Signed-off-by: Dongfeng Yu <dongfengy@nvidia.com>

yibinl-nvidia committed:
[TRTLLM-6420][feat] add support for Eclairv2 model - cherry-pick changes and minor fix (#6493)

Signed-off-by: Yibin Li <109242046+yibinl-nvidia@users.noreply.github.com>

hcyezhang committed:
[None][feat] Core Metrics Implementation (#5785)

Signed-off-by: Ye Zhang <zhysishu@gmail.com>
Signed-off-by: Shunkang <182541032+Shunkangz@users.noreply.github.co>

nv-yilinf committed:
[Perf] Improve Llama4 performance for small max_seqlen cases (#6306)

Signed-off-by: Yilin Fan <206948969+nv-yilinf@users.noreply.github.com>

ziyixiong-nv committed:
[TRTLLM-6637][feat] Resolve KV cache divergence issue (#6628)

Signed-off-by: ziyixiong-nv <219238287+ziyixiong-nv@users.noreply.github.com>

EmmaQiaoCh committed:
[None][infra] Waive test main 0808 (#6751)

Signed-off-by: qqiao <qqiao@nvidia.com>

galagam committed:
[#5048][enhance] AutoDeploy: Optimize prepare_inputs (#6634)

Optimize prepare_inputs routine in AutoDeploy, as part of the effort to reduce the performance gap compared to the default backend.
This PR includes two major fixes, and some other minor tweaks:
1. Avoid back and forth data copies
2. Optimize position ids update by separating the implementation for generation mode and context mode.

Signed-off-by: Suyog Gupta <41447211+suyoggupta@users.noreply.github.com>
Signed-off-by: Gal Hubara Agam <96368689+galagam@users.noreply.github.com>
Co-authored-by: Suyog Gupta <41447211+suyoggupta@users.noreply.github.com>

eopXD committed:
[None][chore][kv cache manager] Dead code elimination, we no longer record/fetch through WindowBlockManager:: mContextBlocksByHash (#6249)

No functional change is intended in this MR.

`WindowBlockManager::mCachedBlocksRoot` is now who is responsible
for the bookkeeping of the `KVCacheBlock`, and the `mNextBlocks` is
now the actual hash map that fetches the block.

The `mEnableHashKey` knob and related hashing is removed.

Signed-off-by: eopXD <yuehtingc@nvidia.com>

shaharmor98 committed:
[TRTLLM-6174][feat] Enable FP32 mamba ssm cache (#6574)

Signed-off-by: Shahar Mor <17088876+shaharmor98@users.noreply.github.com>

pcastonguay committed:
[https://nvbugs/5444937][fix] Fixing kv_cache_event unit test (#6753)

Signed-off-by: Patrice Castonguay <55748270+pcastonguay@users.noreply.github.com>

shaharmor98 committed:
[TRTLLM-6823][doc] Add checkpoint refactor docs (#6592)

Signed-off-by: Shahar Mor <17088876+shaharmor98@users.noreply.github.com>

yechank-nvidia committed:
[None][feat] Support SharedTensor on MultimodalParams (#6254)

Signed-off-by: yechank <161688079+yechank-nvidia@users.noreply.github.com>

zerollzeng committed:
[None][feat] improve dataloading for benchmark_dataset by using batch… (#6548)

Signed-off-by: Zero Zeng <38289304+zerollzeng@users.noreply.github.com>

bo-nv committed:
[https://nvbugs/5431127][fix] Run test_disaggregated_deepseek_v3_lite_fp8_nixl[DeepSeek-V3-Lite-fp8] only on hopper (#6736)

Signed-off-by: Bo Deng <deemod@nvidia.com>

chuangz0 committed:
[None][fix] fix same pp disagg (#6730)

Signed-off-by: Chuang Zhu <111838961+chuangz0@users.noreply.github.com>

Tracin committed:
Add gpt-oss GSM8K test. (#6732)

Signed-off-by: Tracin <10434017+Tracin@users.noreply.github.com>

MrGeva committed:
[None][test] Test trtllm-bench AD vs, PT BEs on H100 single gpu (#6487)

Signed-off-by: Eran Geva <19514940+MrGeva@users.noreply.github.com>
Signed-off-by: Gal Hubara Agam <96368689+galagam@users.noreply.github.com>
Co-authored-by: Gal Hubara Agam <96368689+galagam@users.noreply.github.com>

yiqingy0 committed:
[TRTLLM-5633][infra] Force set changed file diff to empty string for post-merge CI (#6777)

Signed-off-by: Yiqing Yan <yiqingy@nvidia.com>

xinhe-nv committed:
[None][chore] remove closed bugs (#6772)

Signed-off-by: xinhe-nv <200704525+xinhe-nv@users.noreply.github.com>
Co-authored-by: Larry <197874197+LarryXFly@users.noreply.github.com>

EmmaQiaoCh committed:
[None][infra] Waive failed tests on main 0811 (#6778)

Signed-off-by: qqiao <qqiao@nvidia.com>

MartinMarciniszyn committed:
fix: Ensure that Python stub generation works against libnvidia-ml stubs (#6188)

Signed-off-by: Martin Marciniszyn Mehringer <11665257+MartinMarciniszyn@users.noreply.github.com>

byshiue committed:
[TRTLLM-5532][feat] store the block of context request into kv cache (#6683)

Signed-off-by: bhsueh <11360707+byshiue@users.noreply.github.com>

lancelly committed:
[None][doc] Add K2 tool calling examples (#6667)

Signed-off-by: Lanyu Liao <lancelly@users.noreply.github.com>
Co-authored-by: Lanyu Liao <lancelly@users.noreply.github.com>

EmmaQiaoCh committed:
[None][infra] Unwaive an updated case to test (#6791)

Signed-off-by: qqiao <qqiao@nvidia.com>

zhenhuaw-me committed:
[None][chore] always try-catch when clear build folder in build_wheel.py (#6748)

Signed-off-by: Zhenhua Wang <zhenhuaw@nvidia.com>

venkywonka committed:
[TRTLLM-6812][feat] Add standardized GitHub issue templates and disable blank issues (#6494)

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

rakib-hasan committed:
[None][fix] Refactoring to avoid circular import when importing torch models (#6720)

Signed-off-by: Rakib Hasan <rhasan@nvidia.com>

achartier committed:
[None][chore] Find LLM_ROOT and LLM_BACKEND_ROOT dynamically (#6763)

Signed-off-by: Aurelien Chartier <2567591+achartier@users.noreply.github.com>

chang-l committed:
[https://nvbugs/5385987][fix] Fix Qwen2 quantization issue by pinning transformers version (#6673)

Signed-off-by: Chang Liu <9713593+chang-l@users.noreply.github.com>
Signed-off-by: Chang Liu (Enterprise Products) <9713593+chang-l@users.noreply.github.com>

jinyangyuan-nvidia committed:
[None][perf] Improve the performance of online EPLB on Hopper by better overlapping (#6624)

Signed-off-by: Jinyang Yuan <154768711+jinyangyuan-nvidia@users.noreply.github.com>

ziyixiong-nv committed:
[https://nvbugs/5441438][fix] Set correct draft length for the cuda graph dummy request (#6701)

Signed-off-by: ziyixiong-nv <219238287+ziyixiong-nv@users.noreply.github.com>

syuoni committed:
[TRTLLM-2285][feat] Enable guided decoding with CUDA graph padding and draft model chunked prefill (#6774)

Signed-off-by: Enwei Zhu <21126786+syuoni@users.noreply.github.com>

Fridah-nv committed:
[#4403][autodeploy] Refactor: Move more transformations to new inf optimizer, Add quantization_source to factory interface (#6760)

Signed-off-by: h-guo18 <67671475+h-guo18@users.noreply.github.com>
Signed-off-by: Frida Hou <201670829+Fridah-nv@users.noreply.github.com>
Co-authored-by: h-guo18 <67671475+h-guo18@users.noreply.github.com>

sklevtsov-nvidia committed:
[None][feat] CUTLASS MoE FC2+Finalize fusion (#3294)

Signed-off-by: Sergey Klevtsov <sklevtsov@nvidia.com>

lancelly committed:
[TRTLLM-6906][chore] Using pybind to bind functions in thop/attentionOp (#6745)

Signed-off-by: Lanyu Liao <lancelly@users.noreply.github.com>

Shunkangz committed:
[None][fix] Fix attention dp log (#6570)

Signed-off-by: Shunkang <182541032+Shunkangz@users.noreply.github.co>

QiJune committed:
[None][fix] fix ci (#6814)

xinhe-nv committed:
[TRTQA-2920][chore] improve hang tests (#6781)

Signed-off-by: Xin He (SW-GPU) <200704525+xinhe-nv@users.noreply.github.com>

jhaotingc committed:
[https://nvbugs/5438869][fix] Set nvfp4 expert w1 w3 weight scale to the same value if they're not (#6656)

Signed-off-by: Jhao-Ting Chen <jhaotingc@nvidia.com>

nv-yilinf commented:
/bot run

tensorrt-cicd commented:
[PR_Github #15012](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/15012/) [ run ] triggered by Bot

nvchenghaoz committed:
[None][feat] Add GPT OSS support for AutoDeploy (#6641)

Signed-off-by: nvchenghaoz <211069071+nvchenghaoz@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #15012](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/15012/) [ run ] completed with state `FAILURE`
[/LLM/main/L0_MergeRequest_PR pipeline #11339](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/11339/) completed with status: 'FAILURE'

Funatiq committed:
[#6187][feat] add LayerNorm module (#6625)

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq committed:
[None][refactor] Simplify decoder state initialization (#6559)

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #15015](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/15015/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #11341](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/11341/) completed with status: 'FAILURE'

dongxuy04 committed:
[TRTLLM-7008][fix] fix wideEP weights loading and args (#6789)

Signed-off-by: Dongxu Yang <78518666+dongxuy04@users.noreply.github.com>

rakib-hasan committed:
[None][fix] Refactoring input prep to allow out-of-tree models (#6497)

Signed-off-by: Rakib Hasan <rhasan@nvidia.com>

kaiyux committed:
feat: Support custom repo_dir for SLURM script  (#6546)

Signed-off-by: Kaiyu Xie <26294424+kaiyux@users.noreply.github.com>
Co-authored-by: xxi <xxi@nvidia.com>

lfr-0531 committed:
[None][fix] Pre-allocate workspaces for DeepGEMM MoE to avoid frequent cudaFree/cudaMalloc (#6811)

Signed-off-by: Fanrong Li <23290157+lfr-0531@users.noreply.github.com>
Signed-off-by: Yuxian Qiu <142763828+yuxianq@users.noreply.github.com>
Co-authored-by: Yuxian Qiu <142763828+yuxianq@users.noreply.github.com>

yechank-nvidia committed:
[TRTLLM-6772][feat] Multimodal benchmark_serving support (#6622)

Signed-off-by: yechank <161688079+yechank-nvidia@users.noreply.github.com>

mikeiovine committed:
[https://nvbugs/5452167][fix] Fix ngram padding issue (#6837)

Signed-off-by: Mike Iovine <6158008+mikeiovine@users.noreply.github.com>

achartier committed:
[#6530][fix] Fix script when using calibration tensors from modelopt (#6803)

Signed-off-by: Aurelien Chartier <2567591+achartier@users.noreply.github.com>

zhou-yuxin committed:
[https://nvbugs/5412456][fix] Fix an illegal instruction was encountered (#6776)

Signed-off-by: Zhou Yuxin <yuxinz@nvidia.com>

yilin-void committed:
[None][feat] DeepEP LL combine FP4 (#6822)

Signed-off-by: Yilin Zhang <18275976+yilin-void@users.noreply.github.com>

hyukn committed:
[TRTLLM-4501][feat] AutoTuner tuning config refactor and valid tactic generalization. (#6545)

* Generalize the definition of tactics so that users can implement more customizable tactic types, making the configurations clearer for each kernel run. 
* Allow the user not to specify the `gen_tuning_buckets` or the `map_to_tuning_buckets` function.
* Other code refactoring.

Signed-off-by: Yukun He <23156053+hyukn@users.noreply.github.com>

Shixiaowei02 committed:
[TRTLLM-7030][fix] Refactor the example doc of dist-serving (#6766)

Signed-off-by: Shixiaowei02 <39303645+Shixiaowei02@users.noreply.github.com>

PerkzZheng committed:
[TRTLLM-7093][fix] the perf regression to cvt_fp4 kernels (#6851)

Signed-off-by: Perkz Zheng <67892460+PerkzZheng@users.noreply.github.com>

zhenhuaw-me committed:
[https://nvbugs/5412885][doc] Add the workaround doc for H200 OOM (#6853)

Signed-off-by: Zhenhua Wang <4936589+zhenhuaw-me@users.noreply.github.com>

rosenrodt committed:
[https://nvbugs/5378031] [feat] Hopper W4A8 MoE supports ModelOpt ckpt for PyT backend (#6200)

Signed-off-by: Anthony Chang <27950904+rosenrodt@users.noreply.github.com>

EmmaQiaoCh committed:
[None][infra] Waive failed cases on main (#6863)

Signed-off-by: qqiao <qqiao@nvidia.com>

danielafrimi committed:
[None][feat] Support running heterogeneous model execution for Nemotron-H (#6866)

Signed-off-by: Daniel Afrimi <danielafrimi8@gmail.com>

wu6u3tw committed:
[https://nvbugs/5302040][feat] Add whisper support (Bert Attention on SM100 and GPTAttention for cross attention on SM100) (#5527)

Signed-off-by: tinyinl <tinyinl@nvidia.com>

PerkzZheng committed:
[https://nvbugs/5394685][fix] the bug with spec-decoding + SWA && an accuracy issue related to 2CTA MLA (#6834)

Signed-off-by: Perkz Zheng <67892460+PerkzZheng@users.noreply.github.com>

mikeiovine committed:
[https://nvbugs/5410399][chore] Unwaive mtp llmapi test (#6833)

Signed-off-by: Mike Iovine <6158008+mikeiovine@users.noreply.github.com>

Linda-Stadter committed:
[None][fix] max_num_sequences argument in nanobind (#6862)

Signed-off-by: Linda-Stadter <57756729+Linda-Stadter@users.noreply.github.com>

nv-yilinf commented:
/bot run

tensorrt-cicd commented:
[PR_Github #15213](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/15213/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #15213](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/15213/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #11490](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/11490/) completed with status: 'FAILURE'

nv-yilinf committed:
Allocate MoE workspace only when necessary

Signed-off-by: Yilin Fan <206948969+nv-yilinf@users.noreply.github.com>

nv-yilinf commented:
/bot run

tensorrt-cicd commented:
[PR_Github #15346](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/15346/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #15346](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/15346/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #11579](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/11579/) completed with status: 'SUCCESS'
Pipeline passed with automatic retried tests. Check the [rerun report](https://urm.nvidia.com/artifactory/sw-tensorrt-generic/llm-artifacts//LLM/main/L0_MergeRequest_PR/11579/test-results/rerun_report.html) for details.

nv-yilinf commented:
Create this PR because of the branch retarget mess. Created a replica targeting release/1.0 https://github.com/NVIDIA/TensorRT-LLM/pull/6955
